pkg/process/monitor/process_monitor.go

```diff
@@ -185,19 +221,22 @@ func (pm *ProcessMonitor) mainEventLoop() {
 				return
 			}
 
-			pm.eventCount.Inc()
+			if maxChannelSize < len(pm.netlinkEventsChannel) {
+				maxChannelSize = len(pm.netlinkEventsChannel)
+			}
```

Comment:
not sure if that gives us any valuable information.
furthermore I have a concern about the performance impact of that

Reply:
internally chan size is maintained `qcount` https://github.com/golang/go/blob/master/src/runtime/chan.go#L34C1
and len(chan) will call https://github.com/golang/go/blob/master/src/runtime/chan.go#L728